homeassistant/components/lcn/websocket.py

```diff
@@ -0,0 +1,450 @@
+"""LCN Websocket API."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from functools import wraps
+from typing import TYPE_CHECKING, Any, Final
+
+import lcn_frontend as lcn_panel
+import voluptuous as vol
+
+from homeassistant.components import panel_custom, websocket_api
+from homeassistant.components.websocket_api import AsyncWebSocketCommandHandler
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICES,
+    CONF_DOMAIN,
+    CONF_ENTITIES,
+    CONF_ENTITY_ID,
+    CONF_NAME,
+    CONF_RESOURCE,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    ADD_ENTITIES_CALLBACKS,
+    CONF_DOMAIN_DATA,
+    CONF_HARDWARE_SERIAL,
+    CONF_HARDWARE_TYPE,
+    CONF_SOFTWARE_SERIAL,
+    CONNECTION,
+    DOMAIN,
+)
+from .helpers import (
+    DeviceConnectionType,
+    async_update_device_config,
+    generate_unique_id,
+    get_device_config,
+    get_device_connection,
+    get_resource,
+    purge_device_registry,
+    purge_entity_registry,
+    register_lcn_address_devices,
+)
+from .schemas import (
+    ADDRESS_SCHEMA,
+    DOMAIN_DATA_BINARY_SENSOR,
+    DOMAIN_DATA_CLIMATE,
+    DOMAIN_DATA_COVER,
+    DOMAIN_DATA_LIGHT,
+    DOMAIN_DATA_SCENE,
+    DOMAIN_DATA_SENSOR,
+    DOMAIN_DATA_SWITCH,
+)
+
+if TYPE_CHECKING:
+    from homeassistant.components.websocket_api import ActiveConnection
+
+type AsyncLcnWebSocketCommandHandler = Callable[
+    [HomeAssistant, ActiveConnection, dict[str, Any], ConfigEntry], Awaitable[None]
+]
+
+URL_BASE: Final = "/lcn_static"
+
+
+async def register_panel(hass: HomeAssistant) -> None:
```

Comment:
Maybe change the function to `register_panel_and_ws_api` just for clarity?

Reply:
Done